hack/.ci/lib/e2e.sh

```diff
@@ -181,6 +181,22 @@ function gracefully-shutdown-e2es {
   kubectl_cp -n=e2e e2e:/tmp/artifacts -c=wait-for-artifacts "${ARTIFACTS}" || true
 }
 
+function apply-e2e-workarounds-in-all-clusters {
+  for name in "${!WORKER_KUBECONFIGS[@]}"; do
+    if [[ "${WORKER_KUBECONFIGS[$name]}" == "${KUBECONFIG}" ]]; then
+      # Skip if the control plane cluster is also among the worker clusters.
+      worker_as_control_plane="${name}"
+      continue
+    fi
+
+    KUBECONFIG="${WORKER_KUBECONFIGS[$name]}" apply-e2e-workarounds
+  done
+
+  if [[ -z "${worker_as_control_plane+x}" ]]; then
+    KUBECONFIG="${KUBECONFIG}" apply-e2e-workarounds
+  fi
```

Comment:
```suggestion
  apply-e2e-workarounds
  
  for name in "${!WORKER_KUBECONFIGS[@]}"; do
    if [[ "${WORKER_KUBECONFIGS[$name]}" == "${KUBECONFIG}" ]]; then
      # Skip if the control plane cluster is also among the worker clusters.
      continue
    fi

    KUBECONFIG="${WORKER_KUBECONFIGS[$name]}" apply-e2e-workarounds
  done
```
can be simplified a bit, similarly to the existing funcs

Reply:
good idea, done